### PR TITLE
Provision Stage ELK stack

### PR DESCRIPTION
**Pull request by stvstnfrd**
TODO:
- [ ] squash

### CODE REVIEW

**Comment by stvstnfrd** [comment]
@caesar2164 @rachelg4 @caseylitton

File: stanford/elasticsearch.yml [hunk 1]
```diff
@@ -1,7 +1,32 @@
...
+  gather_facts: True
+  vars:
+    # take in number of mega bytes, will scale it properly 
+    ELASTICSEARCH_HEAP_SIZE: "{{ (ansible_memtotal_mb / 2) | int }}m"
+    Number: 1
+    Cluster: elasticsearch-elk
+    Deployment: "{{ Deployment }}"
   vars_files:
     - "{{ secure_dir }}/vars/default/common.yaml"
     - "{{ secure_dir }}/vars/{{ env }}/users.yml"
```
Comment (stvstnfrd) [inline stanford/elasticsearch.yml]: `s/env/Deployment`

File: stanford/logstash.yml [hunk 2]
```diff
@@ -1,15 +1,37 @@
...
-  hosts: tag_role_logs:&tag_environment_{{ env }}
+  hosts: launched
   sudo: True
   gather_facts: True
   vars:
     LOGSTASH_ELASTICSEARCH_HOST: localhost
     # take in number of mega bytes, will scale it properly 
     ELASTICSEARCH_HEAP_SIZE: "{{ (ansible_memtotal_mb / 2) | int }}m"
+    Number: 1
+    KIBANA_NGINX_PORT: 80
```
Comment (stvstnfrd) [inline stanford/logstash.yml]: Can drop; they are defaults.

File: stanford/logstash.yml [hunk 3]
```diff
@@ -1,15 +1,37 @@
...
+    KIBANA_SSL_NGINX_PORT: 443
```
Comment (stvstnfrd) [inline stanford/logstash.yml]: Can drop; they are defaults.

File: stanford/logstash.yml [hunk 4]
```diff
@@ -1,15 +1,37 @@
...
+    Cluster: logstash
+    Deployment: "{{ Deployment }}"
+  vars_files:
+    - "{{ secure_dir }}/vars/{{ Deployment }}/users.yml"
   roles:
     - common
     - oraclejdk
-    - elasticsearch
     - logstash
     - filebeat
```
Comment (stvstnfrd) [inline stanford/logstash.yml]: Instead, do this in `beats.yml`.

File: stanford/kibana.yml [hunk 5]
```diff
@@ -1,18 +1,36 @@
...
+    subnet_cidr: '10.0.110.0/24'
 - name: Configure instance(s)
-  hosts: tag_role_kibana:&tag_environment_{{ env }}
+  hosts: launched
   sudo: True
   gather_facts: True
   vars:
-    KIBANA_SERVER_NAME: "192.168.33.10"
+    Number: 1
     KIBANA_NGINX_PORT: 80
```
Comment (stvstnfrd) [inline stanford/kibana.yml]: Drop ln18,19

File: stanford/playbooks/provision.yml [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+      cidr: "{{ subnet_cidr }}"
+      az: "{{ subnet_az }}"
+      tags:
+      - key: Cluster
+        value: "{{ Cluster }}"
+      - key: Deployment
+        value: "{{ Deployment }}"
+      - key: Name
+        value: "{{ Deployment }}-{{ Cluster }}"
+      route_table_id: "{{ route_table_id }}"
```
Comment (stvstnfrd) [inline stanford/playbooks/provision.yml]: Define these in secrets.

File: stanford/playbooks/provision.yml [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+      network_acl_id: "{{ network_acl_id }}"
+    register: subnet
+  - debug: var=subnet
+    when: ec2.instances | length == 0
+  - name: create security group
+    ec2_group:
+      name: "{{ Deployment }}-group-{{ Cluster }}"
+      description: an example EC2 group
```
Comment (stvstnfrd) [inline stanford/playbooks/provision.yml]: TODO: give a better description.

File: stanford/playbooks/provision.yml [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+      vpc_id: "{{ vpc.vpc_id }}"
+      region: "{{ region }}"
+      rules: "{{ security_group_rules_ingress }}"
```
Comment (stvstnfrd) [inline stanford/playbooks/provision.yml]: TODO: add for logstash and kibana.

File: stanford/playbooks/rotate.yml [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+        region: "{{ region }}"
+        tags:
+          Deployment: "{{ Deployment }}"
+          Cluster: "{{ Cluster }}"
+          Number: "{{ Number|int }}"
+          Name: "{{ Deployment }}-box-{{ Cluster }}-{{ Number }}"
+          todo: deploy
+      register: ec2_new
+    - pause: minutes=3
+    - local_action:
```
Comment (stvstnfrd) [inline stanford/playbooks/rotate.yml]: Add a name

File: stanford/playbooks/rotate.yml [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+        module: ec2_elb_lb
+        name: "{{ Deployment }}-elb-{{ Cluster }}"
+        scheme: internal
+        state: present
+        subnets:
+          - "{{ subnet.subnet_id }}"
+        listeners:
```
Comment (stvstnfrd) [inline stanford/playbooks/rotate.yml]: Move to variable.

File: stanford/playbooks/rotate.yml [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+          - protocol: tcp # options are http, https, ssl, tcp
+            load_balancer_port: 9200
+            instance_port: 9200
+        health_check:
```
Comment (stvstnfrd) [inline stanford/playbooks/rotate.yml]: Move to variable.

File: stanford/playbooks/rotate.yml [hunk 12]
```diff
@@ -0,0 +1,97 @@
...
+          Cluster: "{{ Cluster }}"
+          Number: "{{ Number|int }}"
+          Name: "{{ Deployment }}-box-{{ Cluster }}-{{ Number }}"
+          todo: deploy
+      register: ec2_new
+    - pause: minutes=3
+    - local_action:
+        module: ec2_elb_lb
+        name: "{{ Deployment }}-elb-{{ Cluster }}"
+        scheme: internal
```
Comment (stvstnfrd) [inline stanford/playbooks/rotate.yml]: Parameterize.

File: stanford/playbooks/rotate.yml (hunk 9 above)
Comment (stvstnfrd) [inline stanford/playbooks/rotate.yml]: Move to provision

File: stanford/playbooks/rotate.yml [hunk 13]
```diff
@@ -0,0 +1,97 @@
...
+          ping_protocol: tcp
+          ping_port: 9200
+          response_timeout: 5 # seconds
+          interval: 30 # seconds
+          unhealthy_threshold: 2
+          healthy_threshold: 10
+        region: 'us-west-1'
+      register: elb
+    - debug: var=elb
+    - name: Instance Register
```
Comment (stvstnfrd) [inline stanford/playbooks/rotate.yml]: TODO: better name

File: stanford/playbooks/rotate.yml [hunk 14]
```diff
@@ -0,0 +1,97 @@
...
+        region: 'us-west-1'
+    - name: Tag the new instances
+      with_items: ec2_new.instance_ids
+      ec2_tag:
+        resource: '{{ item }}'
+        region: '{{ region }}'
+        state: present
+        tags:
+          todo: run
+    - name: DeRegister
```
Comment (stvstnfrd) [inline stanford/playbooks/rotate.yml]: TODO: better name

File: stanford/playbooks/provision.yml (hunk 8 above)
Comment (stvstnfrd) [inline stanford/playbooks/provision.yml]: Done

File: stanford/kibana.yml [hunk 15]
```diff
@@ -1,18 +1,60 @@
...
+    elb_scheme: 'internet-facing'
+    elb_listeners:
+      - protocol: https # options are http, https, ssl, tcp
+        load_balancer_port: 443
+        instance_port: 80
+        ssl_certificate_id: "{{ ssl_certificate_id }}"
+        instance_protocol: http
+    elb_health_check:
+      ping_protocol: tcp
+      ping_port: 80
```
Comment (rachelg4) [inline stanford/kibana.yml]: Is this where we will update the health check?
The one to ping the elastic search port is /heartbeat

Comment (stvstnfrd) [inline stanford/kibana.yml]: Yup, let me update that now.

**Pull request merged**